common/src/main/java/com/wynntils/models/leaderboard/LeaderboardModel.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.models.leaderboard;
+
+import com.google.gson.JsonElement;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Model;
+import com.wynntils.core.net.Download;
+import com.wynntils.core.net.UrlId;
+import com.wynntils.models.leaderboard.type.LeaderboardBadge;
+import com.wynntils.models.leaderboard.type.LeaderboardProfile;
+import com.wynntils.models.worlds.event.WorldStateEvent;
+import com.wynntils.models.worlds.type.WorldState;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class LeaderboardModel extends Model {
+    private Map<UUID, LeaderboardProfile> leaderboard = new HashMap<>();
+
+    public LeaderboardModel() {
+        super(List.of());
+
+        updateLeaderboard();
+    }
+
+    public List<LeaderboardBadge> getBadges(UUID id) {
+        Optional<LeaderboardProfile> profile = Optional.ofNullable(leaderboard.get(id));
+
+        if (profile.isEmpty()) return List.of();
+
+        return profile.get().getBadges();
+    }
+
+    // Somewhat arbitrary
```

Comment:
Hm. When did legacy do this? Also, please move events before public methods.

Reply:
Looks like it did it on the world join event